apps/wallet/src/ui/app/shared/transaction-summary/cards/BalanceChanges.tsx

```diff
@@ -6,29 +6,60 @@ import {
 	useFormatCoin,
 	type BalanceChange,
 } from '@mysten/core';
+import { parseStructTag, normalizeSuiObjectId } from '@mysten/sui.js';
+import { useMemo } from 'react';
 
 import { CoinsStack } from './CoinStack';
 import { Card } from '../Card';
 import { OwnerFooter } from '../OwnerFooter';
+import Alert from '_components/alert';
+import { CoinIcon } from '_src/ui/app/components/coin-icon';
+import { useRecognizedPackages } from '_src/ui/app/hooks/useRecognizedPackages';
 import { Text } from '_src/ui/app/shared/text';
 
 interface BalanceChangesProps {
 	changes?: BalanceChangeSummary;
 }
 
-function BalanceChangeEntry({ change }: { change: BalanceChange }) {
+function BalanceChangeEntry({
+	change,
+	isUnRecognizedToken,
+}: {
+	change: BalanceChange;
+	isUnRecognizedToken?: boolean;
+}) {
 	const { amount, coinType } = change;
 	const isPositive = BigInt(amount) > 0n;
-
-	const [formatted, symbol] = useFormatCoin(amount, coinType, CoinFormat.FULL);
+	const [formatted, symbol, queryResult] = useFormatCoin(amount, coinType, CoinFormat.FULL);
 
 	return (
 		<div className="flex flex-col gap-2">
 			<div className="flex flex-col gap-2">
+				{isUnRecognizedToken && (
+					<div className="border-t border-gray-45 pt-2">
+						<Text variant="pSubtitleSmall" weight="normal" color="steel-dark">
+							Coins below are not recognized by <span className="text-hero">Sui Foundation.</span>
+						</Text>
+					</div>
+				)}
 				<div className="flex justify-between">
-					<Text variant="pBody" weight="medium" color="steel-dark">
-						Amount
-					</Text>
+					<div className="flex gap-2">
+						<div className="w-5">
+							<CoinIcon size="sm" coinType={coinType} />
+						</div>
+						<div className="flex max-w-[90px] flex-wrap gap-2 gap-y-1 sm:max-w-full">
+							<Text variant="pBody" weight="semibold" color="steel-darker">
+								{queryResult.data?.name || symbol}
```

Comment:
If we find ourself needing the name, we should update `useFormatCoin` to give it to us. Alternatively, you should call `useCoinMetadata` to get this. Using the queryResult really is discouraged.

Reply:
I initially used useCoinMetadata, but I felt it was redundant since queryResult had it. I will revert it.